rs/boundary_node/rate_limits/canister/confidentiality_formatting.rs

```diff
@@ -0,0 +1,94 @@
+use std::marker::PhantomData;
+
+use crate::{
+    access_control::{AccessLevel, ResolveAccessLevel},
+    types::{OutputConfig, OutputRuleMetadata},
+};
+
+/// Trait for formatting confidential data based on access levels
+pub trait ConfidentialityFormatting {
+    type Input: Clone;
+
+    fn format(&self, value: &Self::Input) -> Result<Self::Input, ConfidentialFormatterError>;
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum ConfidentialFormatterError {
+    #[error("Access denied")]
+    _AccessDenied,
+}
+
+/// A generic confidentiality formatter for various data types
+pub struct ConfidentialityFormatter<A, T> {
+    access_resolver: A,
+    phantom: PhantomData<T>,
+}
+
+impl<A, T> ConfidentialityFormatter<A, T> {
+    pub fn new(access_resolver: A) -> Self {
+        Self {
+            access_resolver,
+            phantom: PhantomData,
+        }
+    }
+}
+
+impl<A: ResolveAccessLevel> ConfidentialityFormatting
+    for ConfidentialityFormatter<A, OutputConfig>
+{
+    type Input = OutputConfig;
+
+    fn format(&self, config: &OutputConfig) -> Result<OutputConfig, ConfidentialFormatterError> {
+        let mut config = config.clone();
+        if self.access_resolver.get_access_level() == AccessLevel::RestrictedRead {
```

Comment:
I would switch this such that the default behavior is the one with the least privilege instead of default is full privilege.

Reply:
didn't understand, i think it is exactly the case now, formatting takes place for the user with least privileged access level